### ISSUE TITLE
Feature: Add a new rule for local link validation only (`mkdocs --strict` + `lychee --offline` behavior)

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#382]
## Scope (only link validation)

This ticket is only about adding **local link validation** to rumdl, inspired by:
- MkDocs `build --strict` link validation behavior
- Lychee `--offline` behavior (no network requests)

Nothing else is included.

## Background

- Issue #279
Showed that integrating `lychee-lib` directly is not a good fit for rumdl due to dependencies and binary size. However, there is still a valuable subset that keeps rumdl lean: offline-only validation of local documentation links.

This proposes implementing:
- Lychee-style `--offline` behavior: validate only local targets and never perform network requests.
- MkDocs-style `validation` configuration (warn/info/ignore, plus `relative_to_docs` for absolute links).

### Offline-only behavior (`lychee --offline`)
Add an offline mode that:
- Validates only links that resolve to local docs/files.
- Never performs any network requests.
- Skips external links (`http(s)://`, `mailto:`, etc.) without attempting to validate them.

### MkDocs-style strictness levels (`mkdocs --strict`)
Add MkDocs-like configuration that controls how each validation category is reported:
- `warn`, `info`, `ignore`
- Special value for absolute links in markdown: `relative_to_docs`

Config categories to implement:
- `validation.nav.omitted_files`
- `validation.nav.not_found`
- `validation.nav.absolute_links`
- `validation.links.not_found`
- `validation.links.anchors`
- `validation.links.absolute_links` (`warn|info|ignore|relative_to_docs`)
- `validation.links.unrecognized_links`

Also support MkDocs shorthand where setting `validation.absolute_links` applies to both `validation.nav.absolute_links` and `validation.links.absolute_links` (same idea for other shared keys).

## Expected behavior

- **Universe of files**: use rumdl's normal file discovery (including include/exclude and gitignore behavior) as the set of local files that can be linked to.
- **`links.not_found`**: local file targets that do not exist in that set.
- **`links.anchors`**: `#fragment` and `file.md#fragment` must match an anchor in the target document.
- **`links.absolute_links`**:
  - `warn|info|ignore`: treat `/foo/bar.md` as an absolute path and report per level.
  - `relative_to_docs`: treat `/foo/bar.md` as relative to `docs_dir` and validate like other local links.
- **`links.unrecognized_links`**: relative links that look like they intended a doc target but do not resolve cleanly (for example `../foo/bar/` when `../foo/bar.md` exists).
- **Nav validation (MkDocs flavor only)**:
  - Parse `mkdocs.yml` nav entries and validate referenced files (`nav.not_found`).
  - Detect files under `docs_dir` that exist but are not referenced in nav (`nav.omitted_files`).
  - Flag absolute links in nav (`nav.absolute_links`).

## Out of scope (explicitly)

- Any network link checking (HTTP status, retries, caching, redirects, auth, GitHub tokens, etc.).
- Integrating `lychee-lib` or pulling in heavy GitHub API dependencies.
- Anything beyond link validation.

## Acceptance criteria

- An offline mode exists and guarantees no network requests.
- MkDocs-style `validation.*` config keys are supported in `.rumdl.toml` and `pyproject.toml` (`[tool.rumdl.validation]`).
- All listed validation categories are implemented with `warn|info|ignore`, plus `relative_to_docs` for `validation.links.absolute_links`.
- Validation is workspace-wide (not per-file only).
- JSON schema is updated to include the new config keys for editor validation/autocomplete.

## References

- MkDocs validation config: https://www.mkdocs.org/user-guide/configuration/#validation
- Lychee offline directory checking guide: https://lychee.cli.rs/guides/getting-started/#check-links-in-directories-but-block-all-network-requests
- Lychee `--offline` CLI flag: https://lychee.cli.rs/guides/cli/#--offline
- Issue #279

**Comment by ryanmerolle** [comment]
Would this help with `mkdocs-gen-files` scripts that dynamically create markdown based on scripts?  For these sort of instances where a mkdocs.yml nav references a path that does not exist except after the script generates the required structure.  The structure passes the `--strict` checks with no issue.

The only approach with rumdl is to ignore the whole file based on a particular rule, which is not ideal.  The other option would be add some feature allowing per line comments to ignore rules for the markdown.yml file, which I did not see in the documentation.

Oh and thanks for this wonderful project!

**Comment by rvben** [comment]
Thanks for the detailed feature request, @pygarap.

Now most of what you've described is already implemented across three existing rules. Here's a mapping:

**Link validation (already available):**

| Requested | Existing Rule | Config |
|---|---|---|
| `links.not_found` | **MD057** | Enabled by default |
| `links.anchors` | **MD051** | Cross-file fragment validation (workspace-wide) |
| `links.absolute_links` | **MD057** | `absolute-links = "ignore"` / `"warn"` / `"relative_to_docs"` |

**Nav validation (already available, MkDocs flavor):**

| Requested | Existing Rule | Config |
|---|---|---|
| `nav.not_found` | **MD074** | `not-found = "warn"` (default) |
| `nav.omitted_files` | **MD074** | `omitted-files = "ignore"` (default) |
| `nav.absolute_links` | **MD074** | `absolute-links = "ignore"` (default) |

**Severity levels:**

Per-rule severity overrides are supported. Combined with `--fail-on`, this gives you full control:

```toml
[MD057]
severity = "info"       # report broken links but don't fail build
absolute-links = "relative_to_docs"  # validate absolute links against docs_dir

[MD051]
severity = "warning"    # fragment validation fails the build

[MD074]
not-found = "warn"
omitted-files = "warn"
```

```bash
rumdl check --fail-on warning  # only fail on warning+ severity
```

**Offline guarantee:** rumdl never makes network requests. All link validation is local filesystem only.

**What's not implemented:**
- `links.unrecognized_links` (fuzzy matching for almost-resolving links, e.g. `../foo/bar/` when `../foo/bar.md` exists). MD057 will flag these as broken links, but won't suggest corrections.
- A unified `[validation]` config section — the functionality is spread across `[MD057]`, `[MD051]`, and `[MD074]`.

Would a documentation improvement help make these features more discoverable? Happy to hear if there are specific gaps in the existing behavior.

**Comment by pygarap** [comment]
@rvben can you add a section or a markdown file to the docs that is dedicated specifically for link validation and nav validation?

**Comment by rvben** [comment]
@ryanmerolle For `mkdocs-gen-files` specifically: since those files only exist after the build plugin runs, rumdl will flag links to them as broken. You have a few options:

1. **Inline suppression** for a section with generated links:
   ```markdown
   <!-- rumdl-disable MD057 -->
   [Generated Reference](api/generated.md)
   [Another Generated Page](api/models.md)
   <!-- rumdl-enable MD057 -->
   ```
2. **Per-file ignores** in `.rumdl.toml` for files that primarily link to generated content:
   ```toml
   [per-file-ignores]
   "docs/reference/**/*.md" = ["MD057"]
   ```
3. **File-level suppression**: `<!-- rumdl-disable-file MD057 -->`

**Comment by rvben** [comment]
Thanks for the suggestion, @pygarap! We added a dedicated guide for this: https://rumdl.dev/link-validation/
